modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowDefinitionScanner.java

```diff
@@ -169,61 +170,54 @@ public void update(File artifact) throws Exception {
    * @return the workflow definition if the given contained a valid one, or null if the file can not be parsed.
    */
   public WorkflowDefinition parseWorkflowDefinitionFile(File artifact) {
-    InputStream stream = null;
-    try {
-      stream = new FileInputStream(artifact);
+    try (InputStream stream = new FileInputStream(artifact)) {
       WorkflowDefinition def = WorkflowParser.parseWorkflowDefinition(stream);
       if (def.getOperations().size() == 0)
         logger.warn("Workflow '{}' has no operations", def.getId());
       return def;
     } catch (Exception e) {
       logger.warn("Unable to parse workflow from file '{}', {}", artifact.getName(), e.getMessage());
       return null;
-    } finally {
-      IOUtils.closeQuietly(stream);
     }
   }
 
-  /**
-   * Gets the workflow definitions with the given id.
-   *
-   * @param id
-   * @return the workflow definition if exist or null
-   */
-  public WorkflowDefinition getWorkflowDefinition(String id) {
-    return installedWorkflows.get(id);
+  public Stream<WorkflowDefinition> getAvailableWorkflowDefinitionsForOrganization(final Organization organization) {
+    return installedWorkflows.keySet().stream()
+            .filter(wfi -> wfi.getOrganization() == null || wfi.getOrganization().equals(organization.getId()))
+            .map(WorkflowIdentifier::getId)
+            .distinct()
+            .map(identifier -> getWorkflowDefinitionOrDefault(new WorkflowIdentifier(identifier, organization.getId())));
   }
 
-  /**
-   * Get the list of installed workflow definitions.
-   *
-   * @return the collection of installed workflow definitions id
-   */
-  public Map<String, WorkflowDefinition> getWorkflowDefinitions() {
-    return installedWorkflows;
+  public WorkflowDefinition getWorkflowDefinitionOrDefault(WorkflowIdentifier workflowIdentifier) {
```

Comment:
Same here - leave "orDefault". I think that's confusing anyway, it's not like we have one general workflow definition... and it's currently the only way to get a workflow definition.